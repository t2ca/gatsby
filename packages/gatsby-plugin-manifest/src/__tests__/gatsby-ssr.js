jest.mock(`fs`, () => {
  return {
    readFileSync: jest.fn().mockImplementation(() => `someIconImage`),
  }
})

const { onRenderBody } = require(`../gatsby-ssr`)

let headComponents
const setHeadComponents = args => (headComponents = headComponents.concat(args))

const ssrArgs = {
  setHeadComponents,
}

describe(`gatsby-plugin-manifest`, () => {
  beforeEach(() => {
    global.__PATH_PREFIX__ = ``
    headComponents = []
  })

  it(`Adds a crossorigin attribute to manifest link tag if provided`, () => {
    onRenderBody(ssrArgs, { crossOrigin: `use-credentials` })
    expect(headComponents).toMatchSnapshot()
  })

  it(`Adds a "theme color" meta tag to head if "theme_color_in_head" is not provided`, () => {
    onRenderBody(ssrArgs, { theme_color: `#000000` })
    expect(headComponents).toMatchSnapshot()
  })

  it(`Does not add a "theme color" meta tag if "theme_color_in_head" is set to false`, () => {
    onRenderBody(ssrArgs, {
      theme_color: `#000000`,
      theme_color_in_head: false,
    })
    expect(headComponents).toMatchSnapshot()
  })

  it(`Add a "theme color" meta tag if "theme_color_in_head" is set to true`, () => {
    onRenderBody(ssrArgs, {
      theme_color: `#000000`,
      theme_color_in_head: true,
    })
    expect(headComponents).toMatchSnapshot()
  })

  it(`Adds "shortcut icon" and "manifest" links and "theme_color" meta tag to head`, () => {
    onRenderBody(ssrArgs, {
      icon: true,
      theme_color: `#000000`,
    })
    expect(headComponents).toMatchSnapshot()
  })

  it(`Adds link favicon tag if "include_favicon" is set to true`, () => {
    onRenderBody(ssrArgs, { icon: true, include_favicon: true })
    expect(headComponents).toMatchSnapshot()
  })

  it(`Does not add a "theme_color" meta tag to head if "theme_color" option is not provided or is an empty string, Adds link favicon if "include_favicon" option is not provided`, () => {
    onRenderBody(ssrArgs, { icon: true })
    expect(headComponents).toMatchSnapshot()
  })

  it(`Does not add a link favicon if "include_favicon" option is set to false`, () => {
    onRenderBody(ssrArgs, { icon: true, include_favicon: false })
    expect(headComponents).toMatchSnapshot()
  })

  it(`doesn't add cache busting if "cache_busting_mode" option is set to none`, () => {
    onRenderBody(ssrArgs, { icon: true, cache_busting_mode: `none` })
    expect(headComponents).toMatchSnapshot()
  })

  it(`Does file name cache busting if "cache_busting_mode" option is set to name`, () => {
    onRenderBody(ssrArgs, { icon: true, cache_busting_mode: `name` })
    expect(headComponents).toMatchSnapshot()
  })

  describe(`Creates legacy apple touch links`, () => {
    it(`Using default set of icons`, () => {
      onRenderBody(ssrArgs, {
        icon: true,
        theme_color: `#000000`,
      })
      expect(headComponents).toMatchSnapshot()
    })

    it(`Using user specified list of icons`, () => {
      onRenderBody(ssrArgs, {
        icon: true,
        theme_color: `#000000`,
        icons: [
          {
            src: `/favicons/android-chrome-48x48.png`,
            sizes: `48x48`,
            type: `image/png`,
          },
          {
            src: `/favicons/android-chrome-512x512.png`,
            sizes: `512x512`,
            type: `image/png`,
          },
        ],
      })
      expect(headComponents).toMatchSnapshot()
    })
  })

  describe(`Does not create legacy apple touch links`, () => {
    it(`If "legacy" options is false and using default set of icons`, () => {
      onRenderBody(ssrArgs, {
        icon: true,
        theme_color: `#000000`,
        legacy: false,
      })
      expect(headComponents).toMatchSnapshot()
    })

    it(`If "legacy" options is false and using user specified list of icons`, () => {
      onRenderBody(ssrArgs, {
        icon: true,
        theme_color: `#000000`,
        legacy: false,
        icons: [
          {
            src: `/favicons/android-chrome-48x48.png`,
            sizes: `48x48`,
            type: `image/png`,
          },
          {
            src: `/favicons/android-chrome-512x512.png`,
            sizes: `512x512`,
            type: `image/png`,
          },
        ],
      })
      expect(headComponents).toMatchSnapshot()
    })
  })

  it(`Creates href attributes using pathPrefix`, () => {
    global.__PATH_PREFIX__ = `/path-prefix`

    onRenderBody(ssrArgs, {
      icon: true,
      theme_color: `#000000`,
    })

    headComponents
      .filter(component => component.type === `link`)
      .forEach(component => {
        expect(component.props.href).toEqual(
          expect.stringMatching(/^\/path-prefix\//)
        )
      })
  })
})
