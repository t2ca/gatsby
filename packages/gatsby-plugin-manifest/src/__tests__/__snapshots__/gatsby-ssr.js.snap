// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gatsby-plugin-manifest Add a "theme color" meta tag if "theme_color_in_head" is set to true 1`] = `
Array [
  <link
    href="/manifest.webmanifest"
    rel="manifest"
  />,
  <meta
    content="#000000"
    name="theme-color"
  />,
  <link
    href="/icons/icon-48x48.png?v=null"
    rel="apple-touch-icon"
    sizes="48x48"
  />,
  <link
    href="/icons/icon-72x72.png?v=null"
    rel="apple-touch-icon"
    sizes="72x72"
  />,
  <link
    href="/icons/icon-96x96.png?v=null"
    rel="apple-touch-icon"
    sizes="96x96"
  />,
  <link
    href="/icons/icon-144x144.png?v=null"
    rel="apple-touch-icon"
    sizes="144x144"
  />,
  <link
    href="/icons/icon-192x192.png?v=null"
    rel="apple-touch-icon"
    sizes="192x192"
  />,
  <link
    href="/icons/icon-256x256.png?v=null"
    rel="apple-touch-icon"
    sizes="256x256"
  />,
  <link
    href="/icons/icon-384x384.png?v=null"
    rel="apple-touch-icon"
    sizes="384x384"
  />,
  <link
    href="/icons/icon-512x512.png?v=null"
    rel="apple-touch-icon"
    sizes="512x512"
  />,
]
`;

exports[`gatsby-plugin-manifest Adds "shortcut icon" and "manifest" links and "theme_color" meta tag to head 1`] = `
Array [
  <link
    href="/icons/icon-48x48.png?v=c3ac975253f3e5b8e06581fbe3334df1b6f7adba"
    rel="shortcut icon"
  />,
  <link
    href="/manifest.webmanifest"
    rel="manifest"
  />,
  <meta
    content="#000000"
    name="theme-color"
  />,
  <link
    href="/icons/icon-48x48.png?v=c3ac975253f3e5b8e06581fbe3334df1b6f7adba"
    rel="apple-touch-icon"
    sizes="48x48"
  />,
  <link
    href="/icons/icon-72x72.png?v=c3ac975253f3e5b8e06581fbe3334df1b6f7adba"
    rel="apple-touch-icon"
    sizes="72x72"
  />,
  <link
    href="/icons/icon-96x96.png?v=c3ac975253f3e5b8e06581fbe3334df1b6f7adba"
    rel="apple-touch-icon"
    sizes="96x96"
  />,
  <link
    href="/icons/icon-144x144.png?v=c3ac975253f3e5b8e06581fbe3334df1b6f7adba"
    rel="apple-touch-icon"
    sizes="144x144"
  />,
  <link
    href="/icons/icon-192x192.png?v=c3ac975253f3e5b8e06581fbe3334df1b6f7adba"
    rel="apple-touch-icon"
    sizes="192x192"
  />,
  <link
    href="/icons/icon-256x256.png?v=c3ac975253f3e5b8e06581fbe3334df1b6f7adba"
    rel="apple-touch-icon"
    sizes="256x256"
  />,
  <link
    href="/icons/icon-384x384.png?v=c3ac975253f3e5b8e06581fbe3334df1b6f7adba"
    rel="apple-touch-icon"
    sizes="384x384"
  />,
  <link
    href="/icons/icon-512x512.png?v=c3ac975253f3e5b8e06581fbe3334df1b6f7adba"
    rel="apple-touch-icon"
    sizes="512x512"
  />,
]
`;

exports[`gatsby-plugin-manifest Adds a "theme color" meta tag to head if "theme_color_in_head" is not provided 1`] = `
Array [
  <link
    href="/manifest.webmanifest"
    rel="manifest"
  />,
  <meta
    content="#000000"
    name="theme-color"
  />,
  <link
    href="/icons/icon-48x48.png?v=null"
    rel="apple-touch-icon"
    sizes="48x48"
  />,
  <link
    href="/icons/icon-72x72.png?v=null"
    rel="apple-touch-icon"
    sizes="72x72"
  />,
  <link
    href="/icons/icon-96x96.png?v=null"
    rel="apple-touch-icon"
    sizes="96x96"
  />,
  <link
    href="/icons/icon-144x144.png?v=null"
    rel="apple-touch-icon"
    sizes="144x144"
  />,
  <link
    href="/icons/icon-192x192.png?v=null"
    rel="apple-touch-icon"
    sizes="192x192"
  />,
  <link
    href="/icons/icon-256x256.png?v=null"
    rel="apple-touch-icon"
    sizes="256x256"
  />,
  <link
    href="/icons/icon-384x384.png?v=null"
    rel="apple-touch-icon"
    sizes="384x384"
  />,
  <link
    href="/icons/icon-512x512.png?v=null"
    rel="apple-touch-icon"
    sizes="512x512"
  />,
]
`;

exports[`gatsby-plugin-manifest Adds a crossorigin attribute to manifest link tag if provided 1`] = `
Array [
  <link
    crossOrigin="use-credentials"
    href="/manifest.webmanifest"
    rel="manifest"
  />,
  <link
    href="/icons/icon-48x48.png?v=null"
    rel="apple-touch-icon"
    sizes="48x48"
  />,
  <link
    href="/icons/icon-72x72.png?v=null"
    rel="apple-touch-icon"
    sizes="72x72"
  />,
  <link
    href="/icons/icon-96x96.png?v=null"
    rel="apple-touch-icon"
    sizes="96x96"
  />,
  <link
    href="/icons/icon-144x144.png?v=null"
    rel="apple-touch-icon"
    sizes="144x144"
  />,
  <link
    href="/icons/icon-192x192.png?v=null"
    rel="apple-touch-icon"
    sizes="192x192"
  />,
  <link
    href="/icons/icon-256x256.png?v=null"
    rel="apple-touch-icon"
    sizes="256x256"
  />,
  <link
    href="/icons/icon-384x384.png?v=null"
    rel="apple-touch-icon"
    sizes="384x384"
  />,
  <link
    href="/icons/icon-512x512.png?v=null"
    rel="apple-touch-icon"
    sizes="512x512"
  />,
]
`;

exports[`gatsby-plugin-manifest Adds link favicon tag if "include_favicon" is set to true 1`] = `
Array [
  <link
    href="/icons/icon-48x48.png?v=c3ac975253f3e5b8e06581fbe3334df1b6f7adba"
    rel="shortcut icon"
  />,
  <link
    href="/manifest.webmanifest"
    rel="manifest"
  />,
  <link
    href="/icons/icon-48x48.png?v=c3ac975253f3e5b8e06581fbe3334df1b6f7adba"
    rel="apple-touch-icon"
    sizes="48x48"
  />,
  <link
    href="/icons/icon-72x72.png?v=c3ac975253f3e5b8e06581fbe3334df1b6f7adba"
    rel="apple-touch-icon"
    sizes="72x72"
  />,
  <link
    href="/icons/icon-96x96.png?v=c3ac975253f3e5b8e06581fbe3334df1b6f7adba"
    rel="apple-touch-icon"
    sizes="96x96"
  />,
  <link
    href="/icons/icon-144x144.png?v=c3ac975253f3e5b8e06581fbe3334df1b6f7adba"
    rel="apple-touch-icon"
    sizes="144x144"
  />,
  <link
    href="/icons/icon-192x192.png?v=c3ac975253f3e5b8e06581fbe3334df1b6f7adba"
    rel="apple-touch-icon"
    sizes="192x192"
  />,
  <link
    href="/icons/icon-256x256.png?v=c3ac975253f3e5b8e06581fbe3334df1b6f7adba"
    rel="apple-touch-icon"
    sizes="256x256"
  />,
  <link
    href="/icons/icon-384x384.png?v=c3ac975253f3e5b8e06581fbe3334df1b6f7adba"
    rel="apple-touch-icon"
    sizes="384x384"
  />,
  <link
    href="/icons/icon-512x512.png?v=c3ac975253f3e5b8e06581fbe3334df1b6f7adba"
    rel="apple-touch-icon"
    sizes="512x512"
  />,
]
`;

exports[`gatsby-plugin-manifest Creates legacy apple touch links Using default set of icons 1`] = `
Array [
  <link
    href="/icons/icon-48x48.png?v=c3ac975253f3e5b8e06581fbe3334df1b6f7adba"
    rel="shortcut icon"
  />,
  <link
    href="/manifest.webmanifest"
    rel="manifest"
  />,
  <meta
    content="#000000"
    name="theme-color"
  />,
  <link
    href="/icons/icon-48x48.png?v=c3ac975253f3e5b8e06581fbe3334df1b6f7adba"
    rel="apple-touch-icon"
    sizes="48x48"
  />,
  <link
    href="/icons/icon-72x72.png?v=c3ac975253f3e5b8e06581fbe3334df1b6f7adba"
    rel="apple-touch-icon"
    sizes="72x72"
  />,
  <link
    href="/icons/icon-96x96.png?v=c3ac975253f3e5b8e06581fbe3334df1b6f7adba"
    rel="apple-touch-icon"
    sizes="96x96"
  />,
  <link
    href="/icons/icon-144x144.png?v=c3ac975253f3e5b8e06581fbe3334df1b6f7adba"
    rel="apple-touch-icon"
    sizes="144x144"
  />,
  <link
    href="/icons/icon-192x192.png?v=c3ac975253f3e5b8e06581fbe3334df1b6f7adba"
    rel="apple-touch-icon"
    sizes="192x192"
  />,
  <link
    href="/icons/icon-256x256.png?v=c3ac975253f3e5b8e06581fbe3334df1b6f7adba"
    rel="apple-touch-icon"
    sizes="256x256"
  />,
  <link
    href="/icons/icon-384x384.png?v=c3ac975253f3e5b8e06581fbe3334df1b6f7adba"
    rel="apple-touch-icon"
    sizes="384x384"
  />,
  <link
    href="/icons/icon-512x512.png?v=c3ac975253f3e5b8e06581fbe3334df1b6f7adba"
    rel="apple-touch-icon"
    sizes="512x512"
  />,
]
`;

exports[`gatsby-plugin-manifest Creates legacy apple touch links Using user specified list of icons 1`] = `
Array [
  <link
    href="/favicons/android-chrome-48x48.png?v=c3ac975253f3e5b8e06581fbe3334df1b6f7adba"
    rel="shortcut icon"
  />,
  <link
    href="/manifest.webmanifest"
    rel="manifest"
  />,
  <meta
    content="#000000"
    name="theme-color"
  />,
  <link
    href="/favicons/android-chrome-48x48.png?v=c3ac975253f3e5b8e06581fbe3334df1b6f7adba"
    rel="apple-touch-icon"
    sizes="48x48"
  />,
  <link
    href="/favicons/android-chrome-512x512.png?v=c3ac975253f3e5b8e06581fbe3334df1b6f7adba"
    rel="apple-touch-icon"
    sizes="512x512"
  />,
]
`;

exports[`gatsby-plugin-manifest Does file name cache busting if "cache_busting_mode" option is set to name 1`] = `
Array [
  <link
    href="/icons/icon-48x48-c3ac975253f3e5b8e06581fbe3334df1b6f7adba.png"
    rel="shortcut icon"
  />,
  <link
    href="/manifest.webmanifest"
    rel="manifest"
  />,
  <link
    href="/icons/icon-48x48-c3ac975253f3e5b8e06581fbe3334df1b6f7adba.png"
    rel="apple-touch-icon"
    sizes="48x48"
  />,
  <link
    href="/icons/icon-72x72-c3ac975253f3e5b8e06581fbe3334df1b6f7adba.png"
    rel="apple-touch-icon"
    sizes="72x72"
  />,
  <link
    href="/icons/icon-96x96-c3ac975253f3e5b8e06581fbe3334df1b6f7adba.png"
    rel="apple-touch-icon"
    sizes="96x96"
  />,
  <link
    href="/icons/icon-144x144-c3ac975253f3e5b8e06581fbe3334df1b6f7adba.png"
    rel="apple-touch-icon"
    sizes="144x144"
  />,
  <link
    href="/icons/icon-192x192-c3ac975253f3e5b8e06581fbe3334df1b6f7adba.png"
    rel="apple-touch-icon"
    sizes="192x192"
  />,
  <link
    href="/icons/icon-256x256-c3ac975253f3e5b8e06581fbe3334df1b6f7adba.png"
    rel="apple-touch-icon"
    sizes="256x256"
  />,
  <link
    href="/icons/icon-384x384-c3ac975253f3e5b8e06581fbe3334df1b6f7adba.png"
    rel="apple-touch-icon"
    sizes="384x384"
  />,
  <link
    href="/icons/icon-512x512-c3ac975253f3e5b8e06581fbe3334df1b6f7adba.png"
    rel="apple-touch-icon"
    sizes="512x512"
  />,
]
`;

exports[`gatsby-plugin-manifest Does not add a "theme color" meta tag if "theme_color_in_head" is set to false 1`] = `
Array [
  <link
    href="/manifest.webmanifest"
    rel="manifest"
  />,
  <link
    href="/icons/icon-48x48.png?v=null"
    rel="apple-touch-icon"
    sizes="48x48"
  />,
  <link
    href="/icons/icon-72x72.png?v=null"
    rel="apple-touch-icon"
    sizes="72x72"
  />,
  <link
    href="/icons/icon-96x96.png?v=null"
    rel="apple-touch-icon"
    sizes="96x96"
  />,
  <link
    href="/icons/icon-144x144.png?v=null"
    rel="apple-touch-icon"
    sizes="144x144"
  />,
  <link
    href="/icons/icon-192x192.png?v=null"
    rel="apple-touch-icon"
    sizes="192x192"
  />,
  <link
    href="/icons/icon-256x256.png?v=null"
    rel="apple-touch-icon"
    sizes="256x256"
  />,
  <link
    href="/icons/icon-384x384.png?v=null"
    rel="apple-touch-icon"
    sizes="384x384"
  />,
  <link
    href="/icons/icon-512x512.png?v=null"
    rel="apple-touch-icon"
    sizes="512x512"
  />,
]
`;

exports[`gatsby-plugin-manifest Does not add a "theme_color" meta tag to head if "theme_color" option is not provided or is an empty string, Adds link favicon if "include_favicon" option is not provided 1`] = `
Array [
  <link
    href="/icons/icon-48x48.png?v=c3ac975253f3e5b8e06581fbe3334df1b6f7adba"
    rel="shortcut icon"
  />,
  <link
    href="/manifest.webmanifest"
    rel="manifest"
  />,
  <link
    href="/icons/icon-48x48.png?v=c3ac975253f3e5b8e06581fbe3334df1b6f7adba"
    rel="apple-touch-icon"
    sizes="48x48"
  />,
  <link
    href="/icons/icon-72x72.png?v=c3ac975253f3e5b8e06581fbe3334df1b6f7adba"
    rel="apple-touch-icon"
    sizes="72x72"
  />,
  <link
    href="/icons/icon-96x96.png?v=c3ac975253f3e5b8e06581fbe3334df1b6f7adba"
    rel="apple-touch-icon"
    sizes="96x96"
  />,
  <link
    href="/icons/icon-144x144.png?v=c3ac975253f3e5b8e06581fbe3334df1b6f7adba"
    rel="apple-touch-icon"
    sizes="144x144"
  />,
  <link
    href="/icons/icon-192x192.png?v=c3ac975253f3e5b8e06581fbe3334df1b6f7adba"
    rel="apple-touch-icon"
    sizes="192x192"
  />,
  <link
    href="/icons/icon-256x256.png?v=c3ac975253f3e5b8e06581fbe3334df1b6f7adba"
    rel="apple-touch-icon"
    sizes="256x256"
  />,
  <link
    href="/icons/icon-384x384.png?v=c3ac975253f3e5b8e06581fbe3334df1b6f7adba"
    rel="apple-touch-icon"
    sizes="384x384"
  />,
  <link
    href="/icons/icon-512x512.png?v=c3ac975253f3e5b8e06581fbe3334df1b6f7adba"
    rel="apple-touch-icon"
    sizes="512x512"
  />,
]
`;

exports[`gatsby-plugin-manifest Does not add a link favicon if "include_favicon" option is set to false 1`] = `
Array [
  <link
    href="/manifest.webmanifest"
    rel="manifest"
  />,
  <link
    href="/icons/icon-48x48.png?v=c3ac975253f3e5b8e06581fbe3334df1b6f7adba"
    rel="apple-touch-icon"
    sizes="48x48"
  />,
  <link
    href="/icons/icon-72x72.png?v=c3ac975253f3e5b8e06581fbe3334df1b6f7adba"
    rel="apple-touch-icon"
    sizes="72x72"
  />,
  <link
    href="/icons/icon-96x96.png?v=c3ac975253f3e5b8e06581fbe3334df1b6f7adba"
    rel="apple-touch-icon"
    sizes="96x96"
  />,
  <link
    href="/icons/icon-144x144.png?v=c3ac975253f3e5b8e06581fbe3334df1b6f7adba"
    rel="apple-touch-icon"
    sizes="144x144"
  />,
  <link
    href="/icons/icon-192x192.png?v=c3ac975253f3e5b8e06581fbe3334df1b6f7adba"
    rel="apple-touch-icon"
    sizes="192x192"
  />,
  <link
    href="/icons/icon-256x256.png?v=c3ac975253f3e5b8e06581fbe3334df1b6f7adba"
    rel="apple-touch-icon"
    sizes="256x256"
  />,
  <link
    href="/icons/icon-384x384.png?v=c3ac975253f3e5b8e06581fbe3334df1b6f7adba"
    rel="apple-touch-icon"
    sizes="384x384"
  />,
  <link
    href="/icons/icon-512x512.png?v=c3ac975253f3e5b8e06581fbe3334df1b6f7adba"
    rel="apple-touch-icon"
    sizes="512x512"
  />,
]
`;

exports[`gatsby-plugin-manifest Does not create legacy apple touch links If "legacy" options is false and using default set of icons 1`] = `
Array [
  <link
    href="/icons/icon-48x48.png?v=c3ac975253f3e5b8e06581fbe3334df1b6f7adba"
    rel="shortcut icon"
  />,
  <link
    href="/manifest.webmanifest"
    rel="manifest"
  />,
  <meta
    content="#000000"
    name="theme-color"
  />,
]
`;

exports[`gatsby-plugin-manifest Does not create legacy apple touch links If "legacy" options is false and using user specified list of icons 1`] = `
Array [
  <link
    href="/favicons/android-chrome-48x48.png?v=c3ac975253f3e5b8e06581fbe3334df1b6f7adba"
    rel="shortcut icon"
  />,
  <link
    href="/manifest.webmanifest"
    rel="manifest"
  />,
  <meta
    content="#000000"
    name="theme-color"
  />,
]
`;

exports[`gatsby-plugin-manifest doesn't add cache busting if "cache_busting_mode" option is set to none 1`] = `
Array [
  <link
    href="/icons/icon-48x48.png"
    rel="shortcut icon"
  />,
  <link
    href="/manifest.webmanifest"
    rel="manifest"
  />,
  <link
    href="/icons/icon-48x48.png"
    rel="apple-touch-icon"
    sizes="48x48"
  />,
  <link
    href="/icons/icon-72x72.png"
    rel="apple-touch-icon"
    sizes="72x72"
  />,
  <link
    href="/icons/icon-96x96.png"
    rel="apple-touch-icon"
    sizes="96x96"
  />,
  <link
    href="/icons/icon-144x144.png"
    rel="apple-touch-icon"
    sizes="144x144"
  />,
  <link
    href="/icons/icon-192x192.png"
    rel="apple-touch-icon"
    sizes="192x192"
  />,
  <link
    href="/icons/icon-256x256.png"
    rel="apple-touch-icon"
    sizes="256x256"
  />,
  <link
    href="/icons/icon-384x384.png"
    rel="apple-touch-icon"
    sizes="384x384"
  />,
  <link
    href="/icons/icon-512x512.png"
    rel="apple-touch-icon"
    sizes="512x512"
  />,
]
`;
